### PR TITLE
feat: Adds possibility to set image pull secrets in all charts

### LINKED ISSUE #203: Please add a way to set imagePullSecrets to deployments
### Summary
Please add a way to set imagePullSecrets to deployments

### Use cases
In busy kubernetes clusters, one can get rate limited to download images on dockerhub (like the default of this helm chart)

### Proposed solution
Simply add a field in the values that would get injected in the deployment

### PR DESCRIPTION
**Pull request by luflow**
Adds possibility of image pull secrets to all charts

on connect: global
on secrets injector: in the injector scope

closes #203

### CODE REVIEW

**Comment by luflow** [comment]
Also fixed some whitespace issues in the two templates I touched.

**Comment by luflow** [comment]
@edif2008 Fixed the linting errors

**Review by edif2008 — COMMENTED** [review]
Thank you so much for your contribution and adding this enhancement! 😊 

I've left a couple of small comments and once they're addressed, I will approve this PR.

**Comment by edif2008** [comment]
I've also noticed just now that your branch seems to be in conflict with the latest `main`. What I recommend is to update your branch with the latest `main` while you address the comments.

File: charts/secrets-injector/templates/deployment.yaml [hunk 1]
```diff
@@ -35,8 +35,14 @@ spec:
...
           lifecycle:
             preStop:
               exec:
                 command: [ "/bin/sh", "-c", "/prestop.sh" ]
+      {{- with .Values.injector.imagePullSecrets }}
+      imagePullSecrets:
+      {{- range . }}
+        - name: {{ . | quote }}
+      {{- end }}
+      {{- end }}
```
[the remark this answers is not in the document]
Comment (luflow) [reply charts/secrets-injector/templates/deployment.yaml]: Fixed

File: charts/connect/templates/connect-deployment.yaml [hunk 2]
```diff
@@ -166,4 +166,10 @@ spec:
...
               subPath: profiler
         {{- end }}
-{{- end }}
+      {{- with .Values.global.imagePullSecrets }}
+      imagePullSecrets:
+      {{- range . }}
+        - name: {{ . | quote }}
+      {{- end }}
+      {{- end }}
+{{- end }}
```
[the remark this answers is not in the document]
Comment (luflow) [reply charts/connect/templates/connect-deployment.yaml]: Fixed

File: charts/connect/values.yaml [hunk 3]
```diff
@@ -5,6 +5,11 @@
 # global common labels, applied to all ressources
 commonLabels: {}
 
+global:
```
[the remark this answers is not in the document]
Comment (luflow) [reply charts/connect/values.yaml]: True. Fixed.

**Comment by luflow** [comment]
@edif2008 thx for your review, fixed all your comments :)

**Comment by volodymyrZotov** [comment]
Hi @luflow 👋. I merged another one PR https://github.com/1Password/connect-helm-charts/pull/170 earlier today, which caused the conflicts with your one 🤦‍♂️. I solved them, but looks like there is a protection rule on your forked repo, so I could not push the changes into your branch.

I'd appreciate if you have a chance resolve the conflicts once again. Or give us permission to push to your branch, so we can help you with that. And finally merge you PR 😃

Sorry for the inconvenience. I appreciate the efforts and the time you put into making this contribution! 👍

**Comment by luflow** [comment]
@edif2008 @volodymyrZotov Done :)

**Comment by luflow** [comment]
@edif2008 @volodymyrZotov anything else needed? :) Looking forward to the merge 🤩

**Review by volodymyrZotov — APPROVED** [review]

**Pull request merged**